app/bin/run-ecs-task/run-task.sh

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+# Run an ECS task using supplied configurations.
+#
+# See README.md for details on how this script works.
+set -o errexit -o pipefail
+
+DIR=$(dirname "${BASH_SOURCE[0]}")
+
+usage() {
+    echo "Usage: ./run-task.sh -a YOUR.NAME -e ENV_NAME -t TASK_NAME"
+    echo "ex: ./run-task.sh -a bob.smith -e test -t db-migrate-up"
+    echo "See the README.md for further details"
+    exit 1
+}
+
+# This parses CLI arguments mapping the letter params to variables
+# Note that if you want to add more params, you need to add the letter
+# to the string next to getopts below
+while getopts ":a:e:t:s:g:" opt; do
+    case $opt in
+        a)
+            AUTHOR="$OPTARG"
+            ;;
+        e)
+            ENV_NAME="$OPTARG"
+            ;;
+        t)
+            TASK_NAME="$OPTARG"
+            ;;
+        s)
+            SUBNETS="$OPTARG"
+            ;;
+        g)
+            SECURITY_GROUPS="$OPTARG"
+            ;;
+        \?)
+            echo "Invalid option -$OPTARG"
+            exit 1
+            ;;
+    esac
+
+    # Error if an argument has no parameter
+    case $OPTARG in
+        -*) 
+            echo "Option $opt needs a valid argument"
+            exit 1
+            ;;
+    esac
+done
+
+# If the required environment variables are missing, print the usage info and error.
+if [ -z "$ENV_NAME" ] || [ -z "$TASK_NAME" ] || [ -z "$AUTHOR" ]; then
+  usage
+fi
+
+# If no security groups were specified, fetch them from AWS. The response is formatted as
+# {
+#   "SecurityGroups": [
+#       {"GroupName": "csv-handler", GroupId: "sg-123456789", ...}  
+# ]}
+if [ -z "$SECURITY_GROUPS" ]; then
+    # The jq command grabs all returned security group IDs and formats it into a JSON array
+    SECURITY_GROUPS=$(aws ec2 describe-security-groups --group-names csv-handler | jq -r --compact-output "[.SecurityGroups[].GroupId]")
```

Comment:
Could these `aws` commands fail if the user has to specify a `--profile` flag when running aws CLI based on their custom setup?

Reply:
They would, you need a working AWS cli + configuration to get this working. Let me find the docs that explain that and add it to the readme.